usl_models/usl_models/atmo_ml/datasets.py

```diff
@@ -227,17 +234,112 @@ def data_generator():
     )
 
     if shuffle:
-        dataset = dataset.shuffle(buffer_size=len(spatiotemporal_file_names))
-
+        dataset = dataset.shuffle(buffer_size=1000)  # Use a fixed buffer size
     dataset = dataset.batch(batch_size)
 
+    # Use fixed sizes for splitting the dataset

```

Comment:
Can we do this split outside of the function so the caller has complete control over how it's split?

Reply:
done